index.js

```diff
@@ -131,6 +131,36 @@ class ScmBase {
     stats() {
         return {};
     }
+
+    /**
+     * Return a unique identifier for the scmUrl
+     * @method getRepoId
+     * @param  {Object}    config        Configuration
+     * @param  {String}    config.scmUrl The scmUrl to generate ID
+     * @param  {String}    config.token  The token used to authenticate to the SCM
+     * @return {Promise}
+     */
+    getRepoId(config) {
+        const result = Joi.validate(config, dataSchema.plugins.scm.getRepoId);
```

Comment:
This code is going through the effort of validating the `config` input, but doesn't seem to be using it anywhere. Should it be passed to `_getRepoId`?